proto/sentry_protos/snuba/v1/endpoint_find_traces.proto

```diff
@@ -0,0 +1,50 @@
+syntax = "proto3";
+
+package sentry_protos.snuba.v1;
+
+import "google/protobuf/timestamp.proto";
+import "sentry_protos/snuba/v1/request_common.proto";
+import "sentry_protos/snuba/v1/trace_item_filter.proto";
+
+message EventFilter {
+  string trace_item_name = 1;
+  TraceItemFilter filter = 2;
+}
+
+message AndTraceFilter {
+  repeated TraceFilter filters = 1;
+}
+
+message OrTraceFilter {
+  repeated TraceFilter filters = 1;
+}
+
+message NotTraceFilter {
+  repeated TraceFilter filters = 1;
+}
+
+message TraceFilter {
+  oneof filter {
+    AndTraceFilter and_filter = 1;
+    OrTraceFilter or_filter = 2;
+    NotTraceFilter not_filter = 3;
+    EventFilter event_filter = 4;
+  }
+}
+
+message FindTracesRequest {
+  RequestMeta meta = 1;
+  TraceFilter filter = 2;
+  PageToken page_token = 3;
```

Comment:
I don't think this page token object is going to quite work. Mostly because it only consists of `TraceItemFilter`s or `offset`s. Which is not sufficient for paginating through traces.

Reply:
That's an interesting point:

In order to paginate the results, the filter could be applied either to the result set of traces or pushed down to the event tables.

In the first case, the filter would be something like `X < trace_id < Y` or `X < end_timestamp < Y`. I don't know if either is a good idea, since it will require building the entire result set in order to filter.

In the second case, the filter would be pushed down to the event tables, which means a `TraceItemFilter` actually does make sense.

Reply:
What you are saying makes sense. Maybe the `PageToken` needs to have `EventFilter` instead of `TraceItemFilter` as its filter attribute. Because as of now since `TraceItemFilter` is not tied to the actual `trace_item_name`

Reply:
I added this to the PageToken, which also required reorganizing the files a bit.